models/bamf_mr_brain_tumor/utils/InverseRegistrationRunner.py

```diff
@@ -0,0 +1,113 @@
+
+        # some sanity checks
+        assert isinstance(in_registration_datas, InstanceDataCollection)
+        assert isinstance(out_datas, InstanceDataCollection)
+        assert len(in_registration_datas) == len(out_datas)
+
+        # filtered collection must not be empty
+        if len(in_registration_datas) == 0:
+            self.v(f"CONVERT ERROR: no data found in instance {str(instance)}.")
+            return None
+
+        process_dir = self.config.data.requestTempDir(label="inverse-processor")
+
+        # conversion step
+        for i, in_data in enumerate(in_registration_datas):
+            in_mat = in_mat_datas.get(i)
+            out_data = out_datas.get(i)
+            in_seg_data = in_seg_datas.get(i)
+
+            # check datatype 
+            if in_data.type.ftype == FileType.NIFTI:                
+                reverse_transformation_matrix = os.path.join(process_dir, f'{str(uuid.uuid4())}_reverse.mat')
+                reverse_transformation_file = os.path.join(process_dir, f'{str(uuid.uuid4())}.nii.gz')
+                # Command to convert the transformation matrices
+                convert_command = [
+                    "convert_xfm",
+                    "-omat",
+                    str(reverse_transformation_matrix),
+                    "-inverse",
+                    str(in_mat.abspath),
+                ]
+                
+                try:
+                    self.v("Converting transformation matrices...",convert_command)
+                    subprocess.run(convert_command, check=True)
+                    self.v("Transformation matrices converted successfully.")
+                except subprocess.CalledProcessError as e:
+                    self.v("Error converting transformation matrices:", e)
+                # Inverse registration using FLIRT with the saved transformation matrix
+                cmd = [
+                    "flirt",
+                    "-in",
+                    str(in_seg_data.abspath),
+                    "-ref",
+                    str(in_data.abspath),
+                    "-out",
+                    str(reverse_transformation_file),
+                    "-init",
+                    str(
+                        reverse_transformation_matrix
+                    ),  # Use the saved transformation matrix for inverse registration
+                    "-cost",
+                    "normmi",
+                    "-dof",
+                    "12",  # 6 degrees of freedom for rigid registration
+                    "-interp",
+                    "nearestneighbour",  # Interpolation method (adjust as needed)
+                    "-applyxfm",
+                ]
+                self.v("inverse transformation flirt...",cmd)
+                subprocess.run(cmd, check=True)
```

Comment:
Must always use `self.subprocess` to guarantee proper capturing. See more: https://github.com/MHubAI/documentation/blob/main/documentation/mhubio/how_to_write_an_mhubio_module.md#running-a-subprocess-from-a-module